sympy/matrices/matrices.py

```diff
@@ -3957,25 +3962,30 @@ def row_insert(self, pos, mti):
         row
         col_insert
         """
-        if pos == 0:
-            return mti.col_join(self)
-        elif pos < 0:
-            pos = self.rows + pos
+        from sympy.matrices.sparse import MutableSparseMatrix
+        i = pos
         if pos < 0:
-            pos = 0
-        elif pos > self.rows:
-            pos = self.rows
-
-        if self.cols != mti.cols:
+            pos = self.rows + pos
+        if pos < -1 or pos > self.rows:
```

Comment:
The part `pos > self.cols` i think is good, because "It performs the same operation as `self.col_insert(pos=self.cols, V)`. I think this to be valid.
If i allow the `pos < 0 or pos >= self.rows` then following  `raises error`
```
>>> M = Matrix(1, 0, [ ])
>>> V = Matrix( [ 8 ] )
>>> M.col_insert(0, V)         # this raises an IndexError
```
this structure so should support i guess `pos > self.row` instead of `pos >= self.rows`.